taca/utils/bioinfo_tab.py

```diff
@@ -154,22 +182,59 @@ def get_ss_projects(run_dir):
     else:
         csvf=open(FCID_samplesheet_origin, 'rU')
         data=DictReader(csvf)
-    
-
-
 
+    proj_n_sample = False
+    lane = False
     for d in data:
         for v in d.values():
-            matches=pattern.search(v)
-            if matches:
-                project_ids.add(matches.group(1))
+            #if project is found
+            if proj_pattern.search(v):
+                projects = proj_pattern.search(v).group(1)
+                #sample is also found
+                samples = sample_pattern.search(v).group(1)
+                proj_n_sample = True
+                
+            #if a lane is found
+            elif lane_pattern.search(v):
+                #In miseq case, writes off a well hit as lane 1
+                lane_inner = re.compile("[A-H]")
+                if lane_inner.search(v):
+                    lanes = 1
+                else:
+                    lanes = lane_pattern.search(v).group(1)
+                lane = True
+         
+        #Populates structure
+        if proj_n_sample and lane:
+            proj_tree[FCID][lanes][samples][projects]
+            proj_n_sample = False
+            lane = False
+    return proj_tree
+
+"""Sends a custom error e-mail
+    :param flag e-mail state
+    :param info variable that describes the record in some way
+"""
+def error_emailer(flag, info):
+    recipients = CONFIG['mail']['recipients']
     
-    return project_ids
-
-
-
-
+    #no_samplesheet: A run was moved back due to QC/BP-Fail. Some samples still passed
+    #failed_run: Samplesheet for a given project couldn't be found
     
-
+    body='Whazzup! TACA is behaving mad whack yo!\n'
```

Comment:
seconded. I have a pretty poor sense of humor when it comes to code and testing.
The reason for that is that sometimes, people not being part of Scilifelab can happen to see these, on occasion. 
Examples would be genologics remoting in our computers for routine work, or a user presentation where you happen to forget that you left your mails open. And I'm talking of things that already happened...